testplan/testing/multitest/entries/assertions.py

```diff
@@ -1201,14 +1204,29 @@ class FixMatchAll(DictMatchAll):
     Similar to DictMatchAll, however dict keys
     will have fix tag info popups on web UI
     """
-    def __init__(
-        self, values, comparisons,
-        tag_weightings=None, description=None, category=None,
-    ):
+    def __init__(self, values, comparisons,
+                 tag_weightings=None, description=None, category=None):
+        """
+        If all input FIX messages are typed, we enable strict type checking.
```

Comment:
User can use DictMatchAll if they want no magic at all. FixMatchAll is customized for compairing FixMessage and untyped match is the default behavior for single FixMatch as well. Changing the legacy behavior, i see no obvious payback rather than create unnecessary bumps for user onboarding.

Reply:
Comparing typed v.s untyped can be very common in a testcase, one could be from log and another from calculation. Supporting such is very much required.

Reply:
I'm not convinced, but if we want to support type laziness then fine.

I would still prefer for the "untyped_fixtag" comparison to be an explicity opt-in rather than just implicitly used in most situations (basically any situation where both sides of the comparison aren't marked as "typed").